### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by frznlogic**
Please don't pull this straight off, talk to me first and give comments ;). 

I'm basically moving the layer magic into a sub directory and adding some documentation of what's happening in the README.md as well as moving all recipes-qt into qt5-layer dependencies (we shouldn't be able to build it without qt5 anyways) as well as removing temporary patches.

I've also merged the fido branch to master as I believe master should equal the latest release, master should be branched into jethro when this cleanup is done IMHO.

### CODE REVIEW

**Comment by rpannek** [comment]
Did you really want to merge it with `master`?

File: conf/bblayers.conf [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  ${BSPDIR}/sources/meta-openembedded/meta-ruby \
+  ${BSPDIR}/sources/meta-bistro \
+  ${BSPDIR}/sources/meta-ivi/meta-ivi \
+"
+
+BBLAYERS_NON_REMOVABLE ?= " \
+  ${BSPDIR}/sources/poky/meta \
+  ${BSPDIR}/sources/poky/meta-yocto \
+  "
+
```
Comment (tobsan) [inline conf/bblayers.conf]: What is this file doing here?

File: conf/local.conf [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+CONF_VERSION = "1"
+
+#IMAGE_FSTYPES = "tar.bz2 ext3.xz"
+#SDIMG_ROOTFS_TYPE = "ext3.xz"
+#SDIMG_COMPRESSION = "xz"
+
+DISABLE_OVERSCAN = "0"
+
+VIDEO_CAMERA = "1"
+
```
Comment (tobsan) [inline conf/local.conf]: This file also does not belong here.

File: setup-environment [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    SDKMACHINE=$SDKMACHINE
+    DISTRO=$DISTRO
+    EULA=$EULA
+EOF
+else
+    echo "Your configuration files at $1 have not been touched."
+fi
+
+clean_up
+
```
Comment (tobsan) [inline setup-environment]: This script also does not belong here.

**Comment by tobsan** [comment]
I made some comments regarding bblayers.conf, local.conf and setup-environment. None of those belong in the layer, so if they are removed I'm more than happy to pull this. Good job on cleanup!

**Pull request merged**